File: service-common/src/main/java/tops/PlayerTopEntry.java
package tops;

import lombok.Getter;
import lombok.Setter;
import user.Stat;

@Setter
@Getter
public class PlayerTopEntry<V> extends TopEntry<Stat, V> {

    private String userName;
    private String displayName;

    public PlayerTopEntry(Stat stat, V value) {
        super(stat, value);
    }

}

File: common/src/main/kotlin/me/reidj/common/top/PlayerTopEntry.kt
package me.reidj.common.top

import me.reidj.common.user.Stat

data class PlayerTopEntry<V>(override var key: Stat, override var value: V): TopEntry<Stat, V>(key, value) {

    lateinit var userName: String
    lateinit var displayName: String
}
